src/object.c

```diff
@@ -80,7 +80,6 @@ robj *createObjectWithKeyAndExpire(int type, void *ptr, const sds key, long long
      * don't need it now. Then we don't need to realloc if it's needed later. */
     if (has_embkey && !has_expire && bufsize >= min_size + sizeof(long long)) {
         has_expire = 1;
-        min_size += sizeof(long long);
```

Comment:
When `has_expire` is set to 1, it affects the minimum size, so even though `min_size` is not used after this point, this code keeps the value correct. This is for the case we would change some code to use `min_size` in the future. If the code would be structured in a way so `min_size` goes out of scope, it would be even better, but since it doesn't, I think it's better to keep the value correct.

The compiler easily detects that the variable is unused after this point and removes the computation.